props/saber_BC_buttons.h

```diff
@@ -477,738 +1478,1218 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
         }
       }
     }
-    // EVENT_THRUST
-      if (mss.y * mss.y + mss.z * mss.z < 16.0 &&
-          mss.x > 14  &&
-          fusor.swing_speed() < 150) {
-        if (millis() - thrust_begin_millis_ > 15) {
-          Event(BUTTON_NONE, EVENT_THRUST);
-          thrust_begin_millis_ = millis();
-        }
-      } else {
+// EVENT_THRUST
+    if (mss.y * mss.y + mss.z * mss.z < 20.0 &&  // slightly more forgiving than 16.0?
+       (fabs(mss.x) > 14) &&  // Check for both positive and negative x-motion
+        fusor.swing_speed() < 150) {
+      if (millis() - thrust_begin_millis_ > 15) {
+        Event(BUTTON_NONE, EVENT_THRUST);
         thrust_begin_millis_ = millis();
       }
-
-    // Mute timer check - play optional mute.wav first.
-    if (mute_timer_.timerCheck()) {
-      DoMute();
+    } else {
+      thrust_begin_millis_ = millis();
+    }
+// Timers
+    // Play optional mute.wav first.
+    if (mute_all_delay_timer_.isTimerExpired()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnOnHelper();
```

Comment:
SetMute returns false if the sound is already muted, but shouldn't we still call TurnOnHelper() in that case?
(same thing two more times below.)

Reply:
I think because SetMute is set back to true on retraction, there's never a chance of the sound already being muted when we get here.
The only time it's already muted would be during a fast_preset change, in which case it just "quickly" remutes.
https://github.com/NoSloppy/ProffieOS-1/blob/0df79d5b72caaa799577914bdced51d40a1bc67f/props/saber_BC_buttons.h#L1658